### PR TITLE
Add support of namespace annotations 

### PR DESCRIPTION
**Pull request by anastasia-malysheva**
Add support of namespace annotations 

Issue
https://github.com/networkservicemesh/cmd-admission-webhook-k8s/issues/126

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -283,6 +294,18 @@ func (s *admissionWebhookServer) createLabelPatch(p string, v map[string]string)
...
+	if err != nil {
+		panic(err.Error())
+	}
+	clientset, err := kubernetes.NewForConfig(c)
+	if err != nil {
+		panic(err.Error())
+	}
+	return clientset
+}
+
```
Comment (denis-tingaikin) [inline main.go:308]:
```suggestion
```

File: main.go [hunk 2]
```diff
@@ -344,8 +367,9 @@ func main() {
 	s.Use(middleware.Recover())
 
 	var handler = &admissionWebhookServer{
-		config: conf,
-		logger: logger.Named("admissionWebhookServer"),
+		config:    conf,
+		logger:    logger.Named("admissionWebhookServer"),
+		clientset: getClientSet(),
```
Comment (denis-tingaikin) [inline main.go:372]:
```suggestion
```

File: main.go [hunk 3]
```diff
@@ -344,8 +367,9 @@ func main() {
...
 	}
```
Comment (denis-tingaikin) [inline main.go:373]:
```suggestion
	}
	_ , clientset, err := k8s.NewVersionedClient()
	if err != nil {
		logger.Fatal(err.Error())	
	}
```

**Review by denis-tingaikin — COMMENTED** [review]
Please re-use our sdk-k8s utils pkg https://github.com/networkservicemesh/sdk-k8s/blob/main/pkg/tools/k8s/kubeutils.go#L50

It will make webhook always up to date with sdk-k8s/k8s.

File: main.go (hunk 3 above)
Comment (anastasia-malysheva) [reply main.go:368]: @denis-tingaikin I've added fixes as you proposed. Please, take a look.

File: main.go [hunk 4]
```diff
@@ -82,6 +84,15 @@ func (s *admissionWebhookServer) Review(in *admissionv1.AdmissionRequest) *admis
...
 	annotation := podMetaPtr.Annotations[s.config.Annotation]
 
+	// use namespace annotation only if resource doesn't have it's own
+	if annotation == "" {
+		namespace, err := s.clientset.CoreV1().Namespaces().Get(ctx, in.Namespace, v1.GetOptions{})
```
Comment (denis-tingaikin) [inline main.go:89]: Please add a timeout for the context here to avoid a potential goroutines leak/improve responsiveness.

Comment (denis-tingaikin) [inline main.go:89]:
```suggestion
		timeoutCtx, cancel := context.WithTimeout(ctx, time.Second)
		defer cancel()
		namespace, err := s.clientset.CoreV1().Namespaces().Get(timeoutCtx, in.Namespace, v1.GetOptions{})
```

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**